pkg/processor/build/builder.go

```diff
@@ -499,8 +501,26 @@ func (b *Builder) resolveFunctionPath(functionPath string) (string, error) {
 		}
 	}
 
-	// if the function path is a URL - first download the file
+	// if the function path is a URL or type is Github - first download the file
+	codeEntryType := b.options.FunctionConfig.Spec.Build.CodeEntryType
+
+	// user has to provide valid url when code entry type is github
+	if !common.IsURL(functionPath) && codeEntryType == githubEntryType {
+		return "", errors.New("Must provide valid URL when code entry type is github or archive")
+	}
+
+	// for backwards compatibility, don't check for entry type url specifically
 	if common.IsURL(functionPath) {
+		if codeEntryType == githubEntryType {
```

Comment:
I know this is small, but since the builder is extremely dense, move as much functionality as you can to a helper. In this case:
```
functionPath, err := b.getFunctionPathFromGithubURL()
...
```